tensordict/tensordict.py

```diff
@@ -2267,7 +2278,7 @@ def set(
         if self._is_memmap is None:
             self._is_memmap = isinstance(value, MemmapTensor)
 
-        present = key in self.keys(include_nested=True)
+        present = key in keys
```

Comment:
We might have to check that `key` is not a tuple length 1 here since collected the keys into a set. `_TensorDictKeysView.__contains__` can handle this case, but `_TensorDictKeysView.__iter__` does not.